src/devices/src/virtio/net/device.rs

```diff
@@ -456,13 +464,10 @@ impl Net {
             self.rx_deferred_frame = false;
             // process_rx() was interrupted possibly before consuming all
             // packets in the tap; try continuing now.
-            self.process_rx()
-        } else if self.rx_deferred_irqs {
-            self.rx_deferred_irqs = false;
```

Comment:
AFAICT, this could have been a bug that you fixed: if inside `signal_used_queue`, `interrupt_evt.write(1)` failed for whatever reason, this line would have marked no more deferred irqs, but the irq would have remain not delivered. With your changes,  `rx_deferred_irqs` is only set to `false` (deferred irqs are marked "handled") *after* successfully signaling the guest. This way, if there's an error in the interrupt event handling, the irq remains there for later retriggering. Did I get that right?

Reply:
To be honest, I didn't think about this initially. I just wanted to make it easier to follow the `rx_deferred_irqs` flag. But yes, looks like it addresses a corner case.